lib/app/features/feed/stories/views/components/story_viewer/components/core/story_gesture_handler.dart

```diff
@@ -3,30 +3,35 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/features/feed/stories/providers/story_pause_provider.dart';
 
 class StoryGestureHandler extends HookConsumerWidget {
   const StoryGestureHandler({
     required this.child,
     required this.onTapLeft,
     required this.onTapRight,
-    required this.onLongPressStart,
-    required this.onLongPressEnd,
     super.key,
   });
 
   final Widget child;
   final VoidCallback onTapLeft;
   final VoidCallback onTapRight;
-  final VoidCallback onLongPressStart;
-  final VoidCallback onLongPressEnd;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final tapPosition = useState<Offset?>(null);
 
     return GestureDetector(
+      behavior: HitTestBehavior.deferToChild, 
       onTapDown: (details) => tapPosition.value = details.globalPosition,
       onTap: () {
+        final isKeyboardVisible = MediaQuery.viewInsetsOf(context).bottom > 0;
+
+        if (isKeyboardVisible) {
+          FocusManager.instance.primaryFocus?.unfocus();
+          return;
+        }
+
         final screenWidth = MediaQuery.sizeOf(context).width;
```

Comment:
`viewInsetOf` could we use this one please?

Reply:
I suppose `viewInsets` is more about keyboard/system UI overlays which aren't relevant for story viewer gesture handling

Reply:
https://github.com/ice-blockchain/ion-app/pull/307/files#diff-6a32c63bc29bdfb6ae5cff5208679c2c09c214a9578eb96dbb63dfd393f47c06L40
sorry for confusing, i tried to add comment but this line however it is a deleted already. LGTM